iotlabcli/helpers.py

```diff
@@ -188,15 +178,30 @@ def __setitem__(self, key, val):
             raise ValueError('Has different values for same key %r' % key)
 
     def add_file(self, file_path):
-        """Add a file to the dictionary. If None, do nothing """
+        """Add a file to the dictionary.
+        :param file_path the path of the file to add
+        :returns the id of the file in the dict
+        If a file with the same basename already exists inside,
+        then prefix with short hash is used
+        if None do nothing
+        """
         if file_path is None:
-            return
-        self[os.path.basename(file_path)] = read_file(file_path, 'b')
+            return None
+        key = os.path.basename(file_path)
+        value = read_file(file_path, 'b')
+        if key in self and self[key] != value:
```

Comment:
This is duplicating the check done in `__setitem__` in a pythonic way, would be better to do something like:

```
try:
    self[key] = value
except ValueError:
    key = '{hash}_{path}`.format(hash=md5(value), path=key)
    self[key] = value
```

    self['{hash}_{path}`.format(hash=md5

Reply:
I agree

Reply:
Fixed